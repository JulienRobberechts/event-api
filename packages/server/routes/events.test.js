const request = require("supertest");

const app = require("../app");
const { mockOpenEventApi } = require("../tests/mock/openEventApi");

const api = request(app);

describe("GET /events", () => {
  beforeEach(() => {
    mockOpenEventApi();
  });
  it("all", async () => {
    const response = await api.get("/events").expect(200);
    expect(response.body).toMatchSnapshot();
  });
  it("type Music", async () => {
    const response = await api.get("/events?type=Music").expect(200);
    expect(response.body).toMatchSnapshot();
  });
  it("ongoing events", async () => {
    const response = await api
      .get("/events?ongoing=true")
      .expect(200);
    expect(response.body).toMatchSnapshot();
  });
  it("ongoing events of type Experience", async () => {
    const response = await api
      .get("/events?type=Experience")
      .expect(200);
    expect(response.body).toMatchSnapshot();
  });
  it("events with type Unknown", async () => {
    const response = await api
      .get("/events?type=Unknown")
      .expect(200);
    expect(response.body).toEqual([]);
  });
  it("country spain", async () => {
    const response = await api.get("/events?country=ESP").expect(200);
    expect(response.body).toMatchSnapshot();
  });
});
