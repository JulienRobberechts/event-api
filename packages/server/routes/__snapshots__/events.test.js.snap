// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /events all 1`] = `
Array [
  Object {
    "end_date": "2025-09-07",
    "name": "Burning Man",
    "start_date": "2020-08-30",
    "type": "Experience",
  },
  Object {
    "end_date": "2020-02-26",
    "name": "Rio Carnival",
    "start_date": "2020-02-21",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-02-28",
    "name": "Ice & snow sculpture festival",
    "start_date": "2019-12-24",
    "type": "Street art",
  },
  Object {
    "end_date": "2020-08-26",
    "name": "La Tomatina",
    "start_date": "2020-08-26",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-02-25",
    "name": "Mardi gras",
    "start_date": "2020-02-25",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-11-02",
    "name": "Dia de los muertos",
    "start_date": "2020-11-01",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-04-27",
    "name": "King’s day",
    "start_date": "2020-04-27",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-03-08",
    "name": "Full Moon Party",
    "start_date": "2020-03-08",
    "type": "Experience",
  },
  Object {
    "end_date": "2020-07-26",
    "name": "Womad (World Of Music, Arts & Dance)",
    "start_date": "2020-07-23",
    "type": "Music",
  },
  Object {
    "end_date": "2020-02-16",
    "name": "World sacred spirit festival",
    "start_date": "2020-02-13",
    "type": "Music",
  },
  Object {
    "end_date": "2020-01-10",
    "name": "Ouidah international voodoo festival",
    "start_date": "2020-01-05",
    "type": "Religious",
  },
  Object {
    "end_date": "2020-12-14",
    "name": "Mevlâna festival",
    "start_date": "2020-12-10",
    "type": "Religious",
  },
  Object {
    "end_date": "2020-04-12",
    "name": "Semana santa",
    "start_date": "2020-04-05",
    "type": "Religious",
  },
  Object {
    "end_date": "2020-05-07",
    "name": "Vesak",
    "start_date": "2020-05-07",
    "type": "Religious",
  },
]
`;

exports[`GET /events country spain 1`] = `
Array [
  Object {
    "end_date": "2020-08-26",
    "name": "La Tomatina",
    "start_date": "2020-08-26",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-04-12",
    "name": "Semana santa",
    "start_date": "2020-04-05",
    "type": "Religious",
  },
]
`;

exports[`GET /events ongoing events 1`] = `
Array [
  Object {
    "end_date": "2020-02-26",
    "name": "Rio Carnival",
    "start_date": "2020-02-21",
    "type": "Street party",
  },
  Object {
    "end_date": "2020-02-28",
    "name": "Ice & snow sculpture festival",
    "start_date": "2019-12-24",
    "type": "Street art",
  },
]
`;

exports[`GET /events ongoing events of type Experience 1`] = `
Array [
  Object {
    "end_date": "2025-09-07",
    "name": "Burning Man",
    "start_date": "2020-08-30",
    "type": "Experience",
  },
  Object {
    "end_date": "2020-03-08",
    "name": "Full Moon Party",
    "start_date": "2020-03-08",
    "type": "Experience",
  },
]
`;

exports[`GET /events type Music 1`] = `
Array [
  Object {
    "end_date": "2020-07-26",
    "name": "Womad (World Of Music, Arts & Dance)",
    "start_date": "2020-07-23",
    "type": "Music",
  },
  Object {
    "end_date": "2020-02-16",
    "name": "World sacred spirit festival",
    "start_date": "2020-02-13",
    "type": "Music",
  },
]
`;
